rest/client.go

```diff
@@ -113,176 +122,178 @@ func (c *client) DeleteNode(ctx context.Context, id uuid.UUID) (err error) {
-		}
-	}
-
-	return origins, nil
+	return response, nil
 }
 
-func (c *client) GetProviderNodeIDs(ctx context.Context, provider string) ([]uuid.UUID, error) {
+func (c *client) GetProviderNodeIDs(ctx context.Context, provider, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error) {
 	request := rest.Get("origin/{provider}/nodes").
 		Assign("provider", provider).
+		Assign("continuation_token", continuationToken).
+		Assign("limit", limit).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNodeIDs
+	var response models.GetNodesByPartialOriginResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return nil, err
-	}
-
-	// Hierarchy swagger does not tell the model generator that it is an array of UUIDs
-	nodeIDs := make([]uuid.UUID, len(response.NodeIds))
-	for _, s := range response.NodeIds {
-		nodeIDs = append(nodeIDs, uuid.UUID(s))
+		return models.GetNodesByPartialOriginResponse{}, err
 	}
 
-	return nodeIDs, nil
+	return response, nil
 }
 
-func (c *client) GetProviderNodeIDsByType(ctx context.Context, provider, originType string) ([]uuid.UUID, error) {
+func (c *client) GetProviderNodeIDsByType(ctx context.Context, provider, originType, continuationToken string, limit int) (models.GetNodesByPartialOriginResponse, error) {
 	request := rest.Get("origin/{provider}/{type}/nodes").
 		Assign("provider", provider).
 		Assign("type", originType).
+		Assign("continuation_token", continuationToken).
+		Assign("limit", limit).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNodeIDs
+	var response models.GetNodesByPartialOriginResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return nil, err
+		return models.GetNodesByPartialOriginResponse{}, err
 	}
 
-	// Hierarchy swagger does not tell the model generator that it is an array of UUIDs
-	nodeIDs := make([]uuid.UUID, len(response.NodeIds))
-	for _, s := range response.NodeIds {
-		nodeIDs = append(nodeIDs, uuid.UUID(s))
-	}
-
-	return nodeIDs, nil
+	return response, nil
 }
 
-func (c *client) GetOriginNodeID(ctx context.Context, origin models.Origin) (uuid.UUID, error) {
+func (c *client) GetOriginNodeID(ctx context.Context, origin models.Origin) (models.GetNodeByOriginResponse, error) {
 	request := rest.Get("origin/{provider}/{type}/{id}/nodes").
 		Assign("provider", origin.Provider).
 		Assign("type", origin.Type).
 		Assign("id", origin.ID).
 		SetHeader("Accept", "application/json")
 
-	var response models.WebmodelsNodeID
+	var response models.GetNodeByOriginResponse
 	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
-		return uuid.EmptyUUID, err
+		return models.GetNodeByOriginResponse{}, err
 	}
 
-	return response.NodeID, nil
+	return response, nil
+}
+
+func (c *client) LockNode(ctx context.Context, id uuid.UUID, recursive bool) error {
+	request := rest.Put("nodes/{node}/lock?recursive={recursive}").
```

Comment:
The URI template supports the following syntax,
```suggestion
	request := rest.Put("nodes/{node}/lock{?recursive}").
```